aea/protocols/generator.py

```diff
@@ -118,45 +122,20 @@ def _extract_specification(self) -> None:
-            else:
-                raise ProtocolSpecificationParseError("Protocol name could not be found.")
-            if "authors" in self.yaml_documents[0]:
-                self.authors = self.yaml_documents[0]["authors"]
-                if type(self.authors) is not str:
-                    raise ProtocolSpecificationParseError("Protocol's 'authors' is not a string.")
-                if self.authors == "":
-                    raise ProtocolSpecificationParseError("Protocol's 'authors' is empty.")
-            if "version" in self.yaml_documents[0]:
-                self.version = self.yaml_documents[0]["version"]
-                if type(self.version) is not str:
-                    raise ProtocolSpecificationParseError("Protocol's 'version' is not a string.")
-                if self.version == "":
-                    raise ProtocolSpecificationParseError("Protocol's 'version' is empty.")
-            if "license" in self.yaml_documents[0]:
-                self.license_config = self.yaml_documents[0]["license"]
-                if type(self.license_config) is not str:
-                    raise ProtocolSpecificationParseError("Protocol's 'license' is not a string.")
-                if self.license_config == "":
-                    raise ProtocolSpecificationParseError("Protocol's 'license' is empty.")
-            if "url" in self.yaml_documents[0]:
-                self.url = self.yaml_documents[0]["url"]
-                if type(self.url) is not str:
-                    raise ProtocolSpecificationParseError("Protocol's 'url' is not a string.")
-                if self.url == "":
-                    raise ProtocolSpecificationParseError("Protocol's 'url' is empty.")
-            if "description" in self.yaml_documents[0]:
-                self.description = self.yaml_documents[0]["description"]
-                if type(self.description) is not str:
-                    raise ProtocolSpecificationParseError("Protocol's 'description' is not a string.")
-                if self.description == "":
-                    raise ProtocolSpecificationParseError("Protocol's 'description' is empty.")
+        if len(self.yaml_documents) >= 2 \
+                and self.yaml_documents[0] is not None \
+                and self.yaml_documents[1] is not None:
+            for field, field_type in BASIC_FIELDS_AND_TYPES.items():
+                if field in self.yaml_documents[0]:
+                    document_field_value = self.yaml_documents[0][field]
+                    self.__setattr__(field, document_field_value)
+                    if type(document_field_value) is not field_type:
+                        raise ProtocolSpecificationParseError("Protocol's '{}' is not of type {}.".format(field, field_type))
+                    if document_field_value == "":
+                        raise ProtocolSpecificationParseError("Protocol's '{}' is empty.".format(field))
+                elif field == "name":
```

Comment:
I would make this an else and raise: `ProtocolSpecificationParseError("Protocol {} could not be found.".format(field))`. We want all basic fields to be present.

Reply:
Done